dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java

```diff
@@ -200,17 +186,20 @@ private OrganisationUnit validateRequestedOrgUnit(String orgUnitUid) throws BadR
     if (orgUnitUid == null) {
       return null;
     }
-
     OrganisationUnit orgUnit = organisationUnitService.getOrganisationUnit(orgUnitUid);
     if (orgUnit == null) {
       throw new BadRequestException("Org unit is specified but does not exist: " + orgUnitUid);
     }
-
     return orgUnit;
   }
 
-  private void validateUser(User user, Program program, ProgramStage programStage)
+  private void validateUser(
+      User user, Program program, ProgramStage programStage, OrganisationUnit requestedOrgUnit)
       throws ForbiddenException {
+    if (user == null) {
```

Comment:
I am confused by this check.
When we have a null user doesn't mean that we are performing the operation as a super user? In other parts of the system that is true, or at least was true.
It seems to me that it is not responsibility of this mapper to check if a user is logged in or not.

Reply:
The way I understood this, the user is a superuser if the condition below is true, which checks that the authorities the user has correspond to the one of a superuser. If the user is null, it will have no authorities and for that reason, it should not be considered a superuser. Can you point me to these other parts of the code where we assume a user is a superuser if null?

Reply:
Yes, this goes back to the scheduler not having a user when running as admin. I think I fixed this now in 2.41 but it will not get backported so ATM it feels best to keep the `null` check.

Reply:
@jbee if this is fixed in 2.41, is it safe to assume a null user is not a superuser anymore? this way we can get rid of some unnecessary logic going forward.

Reply:
I have not tested it and I am unsure about the `User` vs. `CurrentUserDetails`. I do make sure there are `CurrentUserDetails` present when running as admin but I do not use an actual `User`. But that may be the better way to do it. I have to talk to @netroms about this after I am back from vacation.